cmd/controller.go

```diff
@@ -100,17 +101,46 @@ func parseControllerConfig() {
 			log.Error(err, "Can't get k8s client")
 			os.Exit(1)
 		}
+
 		CSINodeDsName := "juicefs-csi-node"
 		if name := os.Getenv("JUICEFS_CSI_NODE_DS_NAME"); name != "" {
 			CSINodeDsName = name
 		}
-		ds, err := k8sclient.GetDaemonSet(context.TODO(), CSINodeDsName, config.Namespace)
+		labelSelector := &metav1.LabelSelector{
+			MatchLabels: map[string]string{
+				"app": "juicefs-csi-node",
+			},
+		}
+
+		pods, err := k8sclient.ListPod(context.TODO(), config.Namespace, labelSelector, nil)
 		if err != nil {
-			log.Error(err, "Can't get DaemonSet", "ds", CSINodeDsName)
+			log.Error(err, "Can't get CSI pods")
 			os.Exit(1)
 		}
-		config.CSIPod = corev1.Pod{
-			Spec: ds.Spec.Template.Spec,
+		var csiPod *corev1.Pod
+		for i := range pods {
+			isCSINodePod := false
+			// Ensure the pod is managed by the expected DaemonSet
+			for _, ownerRef := range pods[i].OwnerReferences {
+				if ownerRef.Kind == "DaemonSet" && ownerRef.Name == CSINodeDsName {
```

Comment:
Maybe it's not create by daemonset?

cc @zwwhdls

Reply:
Yes, this is what I think.
With label `app=juicefs-csi-node` can not 100% guarantee the uniqueness of CSI node Pod,  and if any pod that is not part of juicefs-csi uses “app=juicefs-csi-node”, it can lead to incidents.

Reply:
limit is set to 1, take the first pod, I think that is enough, no need to worry about special cases